app/src/main/java/fr/free/nrw/commons/quiz/QuizActivity.java

```diff
@@ -75,42 +61,45 @@ public void notKnowAnswer(){
      */
     @Override
     public void onBackPressed() {
-        AlertDialog.Builder alert = new AlertDialog.Builder(this);
-        alert.setTitle(getResources().getString(R.string.warning));
-        alert.setMessage(getResources().getString(R.string.quiz_back_button));
-        alert.setPositiveButton(R.string.continue_message, (dialog, which) -> {
-            Intent i = new Intent(QuizActivity.this, QuizResultActivity.class);
-            dialog.dismiss();
-            i.putExtra("QuizResult",score);
-            startActivity(i);
-        });
-        alert.setNegativeButton("Cancel", (dialogInterface, i) -> dialogInterface.dismiss());
-        AlertDialog dialog = alert.create();
-        dialog.show();
+        new AlertDialog.Builder(this)
+            .setTitle(getResources().getString(R.string.warning))
+            .setMessage(getResources().getString(R.string.quiz_back_button))
+            .setPositiveButton(R.string.continue_message, (dialog, which) -> {
+                final Intent intent = new Intent(this, QuizResultActivity.class);
+                dialog.dismiss();
+                intent.putExtra("QuizResult", score);
+                startActivity(intent);
+            })
+            .setNegativeButton("Cancel", (dialogInterface, i) -> dialogInterface.dismiss())
+            .create()
+            .show();
```

Comment:
Followed the builder pattern rather than repeating "alert." and since the local variables aren't used outside of this scope, there were inlined.